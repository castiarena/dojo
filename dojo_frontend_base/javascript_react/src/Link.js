import React from 'react';

const status = {
    NORMAL: 'link-normal',
    HOVERED: 'link-hovered',
};

export default class Link extends React.Component{
    constructor(props) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            className: status.NORMAL
        };
    }

    _onMouseEnter() {
        this.setState({className : status.HOVERED})
    }

    _onMouseLeave() {
        this.setState({className : status.NORMAL})
    }

    render() {
        return (
            <a
            className={this.state.className}
            href={this.props.page}
            onMouseEnter={this._onMouseEnter}
            onMouseLeave={this._onMouseLeave}>
                {this.props.children}
            </a>
        );
    }
}