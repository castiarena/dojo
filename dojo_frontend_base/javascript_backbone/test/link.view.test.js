import Backbone from 'backbone';
import Link from '../src/link.view';


describe('Link tests', () => {
    it('should be in hovered status', () => {
        const model = new Backbone.Model({
            page: 'google.com',
            text: 'Google page'
        });

        const linkView = new Link({model});
        expect(linkView.render().el.innerHTML).toMatchSnapshot();
    });

    it('should be in hovered status', () => {
        const model = new Backbone.Model({
            page: 'google.com',
            text: 'Google page'
        });

        const linkView = new Link({model});
        linkView.mouseEnterHandler();

        expect(linkView.render().el.innerHTML).toMatchSnapshot();
    });

    it('should be in normal status after mouse enter and leave', () => {
        const model = new Backbone.Model({
            page: 'google.com',
            text: 'Google page'
        });

        const linkView = new Link({model});
        linkView.mouseEnterHandler();
        linkView.mouseLeaveHandler();

        expect(linkView.render().el.innerHTML).toMatchSnapshot();
    });

    it('should validate events association with methods of the Link view', () => {
        const model = new Backbone.Model({});
        const linkView = new Link({model});

        const keyEvents = Object.keys(linkView.getEvents());
        const eventsHandlers = keyEvents.map(keyEvent => linkView.getEvents()[keyEvent]);

        const propsNamesAndMethods = Object.getOwnPropertyNames(Link.prototype);

        expect(eventsHandlers
            .every(eventHandler => propsNamesAndMethods.indexOf(eventHandler) !== -1)
        ).toBeTruthy();

    });
});