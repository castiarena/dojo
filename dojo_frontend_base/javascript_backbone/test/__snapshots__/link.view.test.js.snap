// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link tests should be in hovered status 1`] = `
"<a href=\\"google.com\\" class=\\"link link-normal\\">
            Google page
        </a>"
`;

exports[`Link tests should be in hovered status 2`] = `
"<a href=\\"google.com\\" class=\\"link link-hovered\\">
            Google page
        </a>"
`;

exports[`Link tests should be in normal status after mouse enter and leave 1`] = `
"<a href=\\"google.com\\" class=\\"link link-normal\\">
            Google page
        </a>"
`;
